src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -40,27 +40,48 @@ class SetupRecoveryMethod extends Component {
         success: false,
         emailInvalid: false,
         phoneInvalid: false,
-        activeMethods: []
+        activeMethods: [],
+        hasFetchedMethods: false
     }
 
     componentDidMount() {
-        const { loadRecoveryMethods, accountId, router, recoveryMethods } = this.props;
+        const { router, getAccessKeys, getLedgerKey, isNew } = this.props;
         const { method } = router.location;
+        
+        getAccessKeys()
+        getLedgerKey()
 
         if (method) {
             this.setState({ option: method });
         }
 
-        if (recoveryMethods[accountId]) {
-            const confirmed = recoveryMethods[accountId].filter(method => method.confirmed)
+        if (!isNew) {
+            this.setRecoveryMethods()
+        }
+    }
+
+    setRecoveryMethods = () => {
+        if (this.props.recoveryMethods[this.props.activeAccountId]) {
+            const confirmed = this.props.recoveryMethods[this.props.accountId].filter(method => method.confirmed)
             this.setState({ activeMethods: confirmed.map(method => method.kind) });
         } else {
-            loadRecoveryMethods(accountId)
-                .then((payload) => {
-                    const confirmed = payload.data.filter(method => method.confirmed);
-                    this.setState({ activeMethods: confirmed.map(method => method.kind) });
-                })
+            if (!this.state.hasFetchedMethods) {
```

Comment:
nit: hmm, why not just check for `this.stage.activeMethods`? Why is `hasFetchedMethods` needed?